modules/renter/download.go

```diff
@@ -549,8 +611,7 @@ func (r *Renter) managedWaitOnDownloadWork(ds *downloadState) {
 	// If the chunk has completed, perform chunk recovery.
 	if len(cd.completedPieces) == cd.download.erasureCode.MinPieces() {
 		err := cd.recoverChunk()
-		ds.activePieces -= len(cd.completedPieces)
-		cd.completedPieces = make(map[uint64][]byte)
```

Comment:
@DavidVorick could removing this line have other negative consequences? This commit is currently relying on the reference eventually being garbage collected.

As far as I gather from the surrounding code, this line was needed because `len(cd.completedPieces)` was deducted from `ds.activePieces` elsewhere. Since all references to `len(cd.completedPieces)` was replaced with `cd.assignedPieceCount` resetting the map doesn't seem necessary any more.